### PR TITLE
Update CDK deployment code

### PR DESCRIPTION
**Pull request by markdboyd**
- Updates CDK code to deploy application stack itself instead of a code pipeline
- Adds necessary configuration and deployment code updates for deploying to MCP environments
- Updates deployment docs

### CODE REVIEW

File: deployment/cdk/app.py [hunk 1]
```diff
@@ -135,14 +137,26 @@ def __init__(
...
+        if settings.permissions_boundary_name is not None:
+            boundary = iam.ManagedPolicy.from_managed_policy_name(
+                self,
+                'Boundary',
+                settings.permissions_boundary_name
+            )
+            iam.PermissionsBoundary.of(self).apply(boundary)
+
+        if settings.vpc_id is not None:
+            vpc = ec2.Vpc.from_lookup(self, 'VPC', vpc_id=settings.vpc_id)
```
Comment (markdboyd) [inline deployment/cdk/app.py:151]: We can't create VPCs in MCP accounts so we have to look it up

File: deployment/cdk/app.py [hunk 2]
```diff
@@ -126,6 +126,8 @@ def __init__(
         self,
         scope: core.Construct,
         stack_id: str,
+        stack_name: str,
```
Comment (markdboyd) [inline deployment/cdk/app.py:129]: I found the distinction between `stack_id` and `stack_name` less than clear, so there are updates throughout the code for that.

As far as I understand now, `stack_id` is used to name the internal constructs and Cloudformation resources generated by CDK for your deployment, whereas `stack_name` will control the actual Cloudformation stack name of your resources and is generally a more "user friendly" name (e.g. `production-maap-mmt`)

File: deployment/cdk/app.py [hunk 3]
```diff
@@ -254,31 +269,49 @@ def __init__(
...
-            domain_name=f"mmt{hostname_part}.maap-project.org",
-            domain_zone=aws_route53.HostedZone.from_lookup(
-                self, f"{stack_id}-hosted-zone",
-                domain_name="maap-project.org"),
-            redirect_http=True,
-            task_definition=task_definition
+            task_definition=task_definition,
+            load_balancer=load_balancer,
+            listener_port=443,
+            certificate=certificate
```
Comment (markdboyd) [inline deployment/cdk/app.py:305]: Using the `certificate` allows us to create HTTPS load balancers without having to try looking up the hosted zones in other AWS accounts, which isn't possible. 

See https://github.com/aws/aws-cdk/commit/e6c85e4167cdb38ed056eda17b869e179a6dd1c5

File: deployment/README.md [hunk 4]
```diff
@@ -137,31 +103,53 @@ aws ssm put-parameter \
...
+To set values for these settings, create an `.env` file and add a line for the value of
+each setting (with the setting name prefixed by `MMT_STACK_`):
+
+```env
+MMT_STACK_certificate_arn="arn:aws:acm:us-west-2:12345:certificate/abc123"
+```
+
+#### Optional MCP config
+
+For deploying to MCP environments managed by NASA, you will need to specify values in `.env` for the `permissions_boundary_name` and `vpc_id` that should be used:
```
Comment (abarciauskas-bgse) [inline deployment/README.md:142]: Add `Mission Cloud Platform` for ones still learning the lingo :)

File: deployment/README.md [hunk 5]
```diff
@@ -137,31 +103,53 @@ aws ssm put-parameter \
...
 export MMT_STACK_STAGE="dit"
 
 $ npm run cdk deploy -- --require-approval never
 ```
 
-This pipeline will then deploy the application stack that creates a Postgres database, generates a docker image for the application, configures an ECS Task Definition and Service that uses that Task Definition, configures an application load balancer (ALB) to point to the ECS Service, and configures a custom DNS entry for the service.
+The application stack creates a Postgres database, generates a docker image for the application, configures an ECS Task Definition and Service that uses that Task Definition, configures an application load balancer (ALB) to point to the ECS Service, and configures a custom DNS entry for the service.
 
-6. Undeploy (optional)
+### Undeploy (optional)
```
Comment (abarciauskas-bgse) [inline deployment/README.md]: Undeploy is an interesting term - should we say "tear down" ?

File: deployment/cdk/app.py [hunk 6]
```diff
@@ -135,14 +137,26 @@ def __init__(
...
+                self,
+                'Boundary',
+                settings.permissions_boundary_name
+            )
+            iam.PermissionsBoundary.of(self).apply(boundary)
+
+        if settings.vpc_id is not None:
+            vpc = ec2.Vpc.from_lookup(self, 'VPC', vpc_id=settings.vpc_id)
+        else:
+            vpc = ec2.Vpc(self, f"{stack_name}-vpc", max_azs=2)
```
Comment (abarciauskas-bgse) [inline deployment/cdk/app.py]: Any reason we're limiting this to 2 availability zones?

**Review by abarciauskas-bgse — APPROVED** [review]

File: deployment/cdk/app.py (hunk 6 above)
Comment (markdboyd) [reply deployment/cdk/app.py]: 0b47c104c7e5a3a79bc71acbf3c006d0ff897155

File: deployment/README.md (hunk 5 above)
Comment (markdboyd) [reply deployment/README.md]: 0b47c104c7e5a3a79bc71acbf3c006d0ff897155

File: deployment/README.md (hunk 4 above)
Comment (markdboyd) [reply deployment/README.md:142]: 0b47c104c7e5a3a79bc71acbf3c006d0ff897155

**Pull request merged**